### PR TITLE
Task/api 331 immunization model

### PR DESCRIPTION
**Pull request by eclendenning-va**
Resource: http://www.fhir.org/guides/argonaut/r2/StructureDefinition-argo-immunization.html
Story: https://vasdvp.atlassian.net/browse/API-331
- Built out Immunization Model
- Created Immunization sample data
- Added test support for model and bundle

### CODE REVIEW

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Immunization.java [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+  @Pattern(regexp = Fhir.CODE)
+  String language;
+
+  @Valid Narrative text;
+  @Valid List<SimpleResource> contained;
+  @Valid List<Extension> extension;
+  @Valid List<Extension> modifierExtension;
+  @Valid List<Identifier> identifier;
+  @NotNull Status status;
```
Comment (esubocz-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Immunization.java]: needs @Valid

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Immunization.java [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    @NotNull
+    @Min(1)
+    Integer doseSequence;
+
+    String description;
+    @Valid Reference authority;
+    String series;
+
+    @Min(1)
+    Integer seriesDoses;
```
Comment (esubocz-va) [inline argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Immunization.java]: Card is 0...1, why @Min(1)?

Comment (eclendenning-va) [reply argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Immunization.java]: @Min constrains the value of the Integer. seriesDoses is a postiveInt which is defined as any positive integer (e.g. > 0) .

File: argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Immunization.java (hunk 1 above)
Comment (eclendenning-va) [reply argonaut-api/src/main/java/gov/va/api/health/argonaut/api/resources/Immunization.java]: Nice Catch!

File: argonaut-api/src/test/java/gov/va/api/health/argonaut/api/resources/ImmunizationTest.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+import gov.va.api.health.argonaut.api.samples.SampleImmunizations;
+import java.util.Collections;
+import lombok.extern.slf4j.Slf4j;
+import org.junit.Test;
+
+@Slf4j
+public class ImmunizationTest {
+
+  private final SampleImmunizations immunizationData = SampleImmunizations.get();
+  private final SampleDataTypes dataTypes = SampleDataTypes.get();
```
Comment (esubocz-va) [inline argonaut-api/src/test/java/gov/va/api/health/argonaut/api/resources/ImmunizationTest.java]: With new structure, these aren't both necessary. A declaration like:
private final SampleImmunizations data = SampleImmunizations.get(); will work. & references to each can be replaced by data. (you can use other resource tests for reference)

Comment (eclendenning-va) [reply argonaut-api/src/test/java/gov/va/api/health/argonaut/api/resources/ImmunizationTest.java]: Fixed!

**Pull request merged**